democracy_club/apps/donations/views.py

```diff
@@ -7,19 +8,15 @@ class DonateFormView(TemplateView):
     template_name = "donate.html"
 
 
-class DonateThanksView(TemplateView):
-
-    def get(self, request, *args, **kwargs):
-        resource_uri = request.GET.get('resource_uri')
-        resource_id = request.GET.get('resource_id')
-        resource_type = request.GET.get('resource_type')
-        if resource_id and resource_type and resource_uri:
-            GoCardlessHelper().confirm_payment(
-                resource_uri,
-                resource_id,
-                resource_type
-            )
-        return super(DonateThanksView, self).get(request, *args, **kwargs)
+class ProcessDonationView(RedirectView):
+    def get_redirect_url(self, *args, **kwargs):
+        redirect_flow_id = self.request.GET.get('redirect_flow_id')
```

Comment:
What happens if someone arrives at this view without this url param set? Should this specify a default of `False` or `None`?

Reply:
`.get()` there will default to `None` if it's not provided (`request.GET` is a dict-like object so `.get()` will act like `dict().get()`).

If it's not provided the user just gets redirected to the 'thanks' template view.

Note that the user only requests this URL as part of the redirect from GoCardless, and are then redirected to the thanks page, so they should never see the URL at all, and it wont be linked to from anywhere. Anyone requesting that URL would be doing something…odd anyway.